pkg/controller/update_cluster.go

```diff
@@ -308,6 +309,78 @@ func (c *Controller) addPodToPlacement(cluster *myspec.M3DBCluster, pod *corev1.
 	return nil
 }
 
+func (c *Controller) checkPodsForReplacement(
+	cluster *myspec.M3DBCluster,
+	pods []*corev1.Pod,
+	pl placement.Placement) (string, *corev1.Pod, error) {
+
+	var instancePodID myspec.PodIdentity
+
+	insts := pl.Instances()
+	sort.Sort(placement.ByIDAscending(insts))
+
+	sortedPods, err := sortPods(pods)
+	if err != nil {
+		return "", nil, fmt.Errorf("cannot sort pods: %v", err)
+	}
+
+	for _, pod := range sortedPods {
+
+		clusterPodID, err := c.podIDProvider.Identity(pod.pod, cluster)
+		if err != nil {
+			return "", nil, err
+		}
+
+		for _, inst := range insts {
+			if strings.EqualFold(strings.Split(inst.Hostname(), ".")[0], pod.pod.Name) {
+				if err = json.Unmarshal([]byte(inst.ID()), &instancePodID); err != nil {
+					return "", nil, err
+				}
+
+				if !strings.EqualFold(clusterPodID.UID, instancePodID.UID) {
```

Comment:
Since I think we're going to assume we always have `Name` but might not have `UID` based on user config, can we check `Name` here instead?

Reply:
if we check only `Name` though how do we know the pod is a valid case for replacement?